modules/commons/src/main/frontend/src/components/DropdownsDatePicker.jsx

```diff
@@ -0,0 +1,295 @@
+//OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//SOFTWARE.
+//
+
+import React, { useState } from "react";
+import { Select, MenuItem,  makeStyles } from "@material-ui/core";
+import CalendarTodayIcon from '@material-ui/icons/CalendarToday';
+import PropTypes from "prop-types";
+import moment from "moment";
+
+const DropdownDate = {
+  year: 'year',
+  month: 'month',
+  day: 'day',
+}
+
+const getDaysInMonth = (year, month) => {
+  if (month < 0 || year < 0) return 31;
+  year = +(year);
+  month = +(month) + 1;
+  return new Date(year, month, 0).getDate();
+};
+
+const useStyles = makeStyles(theme => ({
+  select : {
+    paddingLeft: theme.spacing(1),
+  },
+  icon : {
+    float: "right"
+  },
+  default : {
+    opacity: "50%"
+  },
+  container : {
+    borderBottom: "1px solid " + theme.palette.grey["800"],
+  },
+}));
+
+
+/**
+ * React-based date picker. Select date from Day, Month and Year dropdowns.
+ * Code adapted & optimized from https://github.com/ssxv/react-dropdown-date
+ *
+ * @param {string} startDate optional, if not provided 1900-01-01 is startDate, 'yyyy-mm-dd' format only
+ * @param {string} endDate optional, if not provided current date is endDate, 'yyyy-mm-dd' format only
+ * @param {string} selectedDate optional, if not provided default values will be displayed, 'yyyy-mm-dd' format only
+ * @param {array} order optional, Order of the dropdowns
+ * @param {func} onDateChange optional, Callback on day, month or year change
+ * @param {bool} yearReverse optional, If true, the year dropdown is ordered in time reverse order
+ * @param {bool} disabled optional, If true, all dropdowns are disabled
+ * @param {bool} monthShort optional, If true, months options are listed as 3 first characters
+ * @param {bool} formatDate optional, If true, formats date according to the provided `order`
+ * @param {bool} autoFocus optional, If true, autofocus of the first select
+ *
+ */
+
+function DropdownsDatePicker(props) {
+  const { startDate, endDate, selectedDate, order, onDateChange, yearReverse, disabled, monthShort, formatDate, autoFocus, ...rest } = props;
+  const classes = useStyles();
+
+  const sDate = new Date(startDate);
+  const eDate = endDate ? new Date(endDate) : new Date();
+  const selDate = selectedDate ? new Date(selectedDate) : null;
+
+  const startYear = sDate.getFullYear();
+  const startMonth = sDate.getMonth();
+  const startDay = sDate.getDate();
+  const endYear = eDate.getFullYear();
+  const endMonth = eDate.getMonth();
+  const endDay = eDate.getDate();
+
+  const [ selectedYear, setSelectedYear ] = useState( selDate ? selDate.getFullYear() : -1);
+  const [ selectedMonth, setSelectedMonth ] = useState( selDate ? selDate.getMonth() : -1);
+  const [ selectedDay, setSelectedDay ] = useState( selDate ? selDate.getDate() : -1);
+
+
+  let generateYearOptions = () => {
+    let yearOptions = [];
+    yearOptions.push(<MenuItem key={-1} value="-1">yyyy</MenuItem>);
```

Comment:
"yyyy" looks awkward at the to of the list, I would remove the label completely.
```suggestion
    yearOptions.push(<MenuItem key={-1} value="-1"></MenuItem>);
```